### ISSUE TITLE
Convert to use PyPI `pid` instead of `python-pidfile`

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#3037]
Convert to use PyPI `pid` instead of `python-pidfile`.

It appears that an RPM exists for `pid` in Fedora packaging, `python3-pid` (at least as of Fedora 36) so we won't have to carry a `pip3` install package for it.

**Comment by portante** [comment]
While that may be true, the `pid` project is a year more active upstream (Nov 2020) than the `pidfile` project (Dec 2019, aside from @vishalvvr's recent license PR), and the `pid` project has more stars and forks on GitHub that the other.

**Comment by webbnh** [comment]
My concern is that we support more than just the latest version of Fedora:  we support at least one older version, as well as CentOS Stream and RHEL.  So, the fact that the RPM is now available for the latest Fedora isn't all that impressive if we don't have it in most of the other places that we want to go (and, that's equally true for `pid` and `pidfile`...).

**Comment by portante** [comment]
FWIW, `python3-pid` is already shipped with RHEL 8.  I doubt we'll be able to convince RHEL to take in another package like `pid` that is already there.

**Comment by pravins** [comment]
Right, getting into RHEL, is very difficult, but we can get this package in epel. 
Getting into Fedora 35, is also easy, just need to request that branch. 

Off course, we can move pidfile to pid, this is not a big task as well. 
As far as i think, its low priority now due to availability of python3-pid package.
